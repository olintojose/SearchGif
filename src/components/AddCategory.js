import React, { useState } from 'react';

import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {
    const [InputValue, setInputValue] = useState('')
    const handleInputChange= (e) => {
        setInputValue(e.target.value)
        console.log('HandreInputChange llamado')
    }

  
    
  
    const handleSubmit = (e) => {   
        e.preventDefault();
        console.log('handleSubmit');
        //console.log('submit hecho')
       // setCategories( [...categories,'HunterXHunter'] )  No tengo acceso a categories
       if ( InputValue.trim().length > 2) {
        setCategories ( cats => [ InputValue , ...cats ]);
        setInputValue('');
       }
      
    }
    return (
        <form onSubmit={ handleSubmit }>
          <input 
            type="text"
            value={ InputValue }
            onChange={handleInputChange}
          /> 
        </form>
    )
}


AddCategory.propTypes={
    setCategories: PropTypes.func.isRequired
}