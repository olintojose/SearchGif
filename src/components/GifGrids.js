import React from 'react'
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs'
 import { GifGridItem } from './GifGridItem';

export const GifGrids = ({category }) => {


    const  { data:images, loading }= useFetchGifs(category);


    return (
        <>
        <h3 className="animate__animated animate__bounce">{category}</h3>

        <div className="card-grid animate__animated animate__fadeIn">
          
          { loading && <p className="animate__animated animate__flash">Loading</p> }
            
               {
                   images.map ((img ) => (
                      <GifGridItem  
                      key={ img.id }
                         { ...img } 
                      />
                   ))
               }
            

        </div>
        </>
    )
}

GifGrids.propTypes= {
    category: PropTypes.string.isRequired
}
