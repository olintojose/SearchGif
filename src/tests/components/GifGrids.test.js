import React from 'react'
import '@testing-library/jest-dom';

import { shallow } from "enzyme"
import { GifGrids } from "../../components/GifGrids"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')


describe('Pruebas en <GifGrids />', () => {

    const category = 'Goku';
           

    test('Debe mostrar el componente GifGrids correctamente', () => {


        const gifs = [];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        })

        const wrapper = shallow(<GifGrids category={ category }/> )  
        expect (wrapper).toMatchSnapshot();
        
    })

    test('Debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrids category={ category }/> )
       // expect (wrapper).toMatchSnapshot();

       expect( wrapper.find('p').exists()).toBe(false);
       expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
    })
    
})