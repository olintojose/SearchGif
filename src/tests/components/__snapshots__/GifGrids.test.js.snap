// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrids /> Debe mostrar el componente GifGrids correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Goku
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading
    </p>
  </div>
</Fragment>
`;
